src/backend/ceph.rs

```diff
@@ -875,10 +798,18 @@ impl CephBackend {
         let mut part1: String = dev_path.to_string_lossy().to_string();
         part1.push_str("1");
         let part1 = Path::new(&part1);
-        block_utils::unmount_device(&part1)?;
+        let osd_dir = Path::new("/var/lib/ceph/osd/").join(&format!("ceph-{}", osd_id));
+        // check if the device path exists (partition may have been deleted)
+        if part1.exists() {
+            // check if the osd_dir is mounted (might not be if partitions have been deleted)
+            if block_utils::is_mounted(&osd_dir)? {
```

Comment:
block_utils doesn't check if this directory exists before trying to work on it.  This could potentially result in an error

Reply:
made it an if let Ok(true)